spec.md

```diff
@@ -297,6 +297,29 @@ circuit:
 Layer blocks will be compiled to modules whose ports are derived from what they capture from their visible scope.
 For full details of the way layers are compiled, see the FIRRTL ABI specification.
 
+## Options
+
+The option mechanism declares configurable parameters with a pre-defined set of values.
+Options are primarily intended to enable target-specific (ASIC or FPGA) lowerings, but they can also be used specify parameters which are present in the output and can be selected after emission (for example, to tweak a design for size or performance).
+Specialization can occur either in the compiler or in the output via the ABI. For full details, consult the FIRRTL ABI.
```

Comment:
You can collapse both of these sentences together. Options exist to enable specialization. Specialization may then happen before or after Verilog emission.

I am concerned about how to handle the specialization-during-compilation case as this seems to violate an ABI guarantee.